src/QSymbolicsBase/predefined.jl

```diff
@@ -205,23 +205,28 @@ function Base.show(io::IO, x::SProjector)
     print(io,"]")
 end
 
-"""Dagger a Ket into Bra."""
-@withmetadata struct SDagger <: Symbolic{AbstractBra}
-    ket::Symbolic{AbstractKet}
+"""Dagger, i.e., adjoint of quantum objects (kets, bras, operators, and inner products)"""
+@withmetadata struct SDagger{T<:Union{Complex, QObj}} <: Symbolic{Union{Complex, T}}
+    obj
 end
 istree(::SDagger) = true
 arguments(x::SDagger) = [x.ket]
 operation(x::SDagger) = dagger
 exprhead(x::SDagger) = :dagger
-dagger(x::Symbolic{AbstractKet}) = SDagger(x)
+dagger(x::Symbolic{AbstractKet}) = SBra(x.name, x.basis)
```

Comment:
A ket might not be a simple named ket, rather some more sophisticated object. This method, as defined, makes sense for `SKet`, but not for any other symbolic ket types.